trieste/acquisition/rule.py

```diff
@@ -94,27 +88,42 @@ def __init__(
         self,
         builder: AcquisitionFunctionBuilder | None = None,
         optimizer: AcquisitionOptimizer[SP_contra] | None = None,
+        num_query_points: int = 1,
```

Comment:
hmmm ... i've just noticed that if the optimizers accept the batch size, users can need to specify it twice:
```python
EfficientGlobalOptimization(EI().using(""), batchify(optimize_discrete, 3), 3)
```

Reply:
then there's the question: how can users only specify the batch size and not all the other stuff? One option is union types, but a more principled option is a class method

Reply:
would you rather not have batch size as an input to the rule? That would work

Reply:
it's an option. I'm not sure what the best approach for the notebooks would be. For now I'd probably specify the optimizer just cos it would be quicker and we're not rolling in time

Reply:
I think leave it as is. Its worth being able to choose batch size when defining the EGO rule. I think batch size is a property of the BO routine, not the optimizer. 

What do you think @vpicheny ?

Reply:
the batch size should definitely be defined by the rule. It is not great to have to specify it for the optimizer as well.

Reply:
You only need to specify it for the optimizer if you choose to tell it a specific optimizer (which the vast majority of users will not need to do).

Reply:
it's probably fine then? we probably need a good docstring to make this clear though

Reply:
Perhaps we should always call batchify, so that a user can specify a particular type of optimizer (i.e. random search or local or genetic algorithms), but all the batching is dealt with under the hood. 

This is the way that all other BO packages work. It feels awkward to have to specify a batch optimizer. The batchness (and batch size) is just a property of the acquisition function.

Reply:
I vote for always calling batchify (see new PR).  This greatly simplifies the specification of custom optimizers. 

This will also allow for an easy implementation of greedy batches!

Reply:
i don't really mind, though I don't follow how this will make it easier to use greedy batches. I would have thought it will make it harder cos you can't customize which function does the batching